### ISSUE TITLE
Login logic - facebook

### ISSUE THREAD

**Issue by navaru** [outmoded/postmile-web#1]
```
 > git clone git@github.com:hueniverse/postmile.git
 > cd postmile
 > npm update
 > cp config.js.example config.js
 > vi config.js
```

In config.js I added only credentials for Facebook, and I get this error when running `node index.js` and accessing localhost:8000/auth/facebook:

```
Debug: hapi, internal, implementation, error
    TypeError: Uncaught error: Cannot read property 'query' of undefined
    at facebook (/code/postmile/node_modules/postmile-web/lib/login.js:164:21)
    at entry (/code/postmile/node_modules/postmile-web/lib/login.js:71:30)
    at Object.exports.auth (/code/postmile/node_modules/postmile-web/lib/login.js:404:5)
    at /code/postmile/node_modules/hapi/lib/handler.js:95:35
    at protect (/code/postmile/node_modules/hapi/lib/ext.js:100:9)
    at /code/postmile/node_modules/hapi/lib/handler.js:92:9
    at Function.internals.Ext.runProtected (/code/postmile/node_modules/hapi/lib/ext.js:103:5)
    at Object.internals.handler (/code/postmile/node_modules/hapi/lib/handler.js:69:9)
    at exports.execute (/code/postmile/node_modules/hapi/lib/handler.js:31:19)
    at /code/postmile/node_modules/hapi/lib/request.js:311:13
```

In `/lib/login.js` on line 164 (`if (!request.query.code) {`) `request` becomes undefined somewhere along the process. I so there are some inconsistencies, on line 165 is redefined and on line 175 the old object should be available. I wanted to fix it and submit a PR, but I'm just learning hapi 2.0 and its a little hard to follow the logic.

**Comment by hueniverse** [comment]
I haven't testing Facebook login or Yahoo login in 2 years... I want to replace that code with Travelogue.

**Comment by hueniverse** [comment]
Works now using bell.
